CONTRIBUTING.txt

```diff
@@ -0,0 +1,99 @@
+FreeCAD Contribution Process (FCP)
+==================================
+
+FreeCAD's contribution process is inspired by the Collective Code Construction Contract which itself is an evolution of the github.com Fork and Pull Model.
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED",  "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119.
+
+
+0. Status
+
+FreeCAD is in a transition period. The following are to be regarded as GUIDELINES for contribution submission and acceptance. For historical reasons, the actual process MAY diverge from this process during the transition.  Such deviations SHOULD be noted and discussed whenever possible.
+
+1. Goals
+
+    The FreeCAD Contribution Process is expressed here with the following specific goals in mind:
+
+    1. To provide transparency and fairness in the contribution process.
+    2. To allow useful contributions to be included as quickly as possible.
+    3. To preserve and improve the code quality while encouraging appropriate experimentation and risk-taking.
+    4. To minimize dependence on individual Contributors by encouraging a large pool of active Contributors.
+    5. To be inclusive of many viewpoints and to harness a diverse set of skills.
+    6. To provide an encouraging environment where Contributors learn and improve their skills.
+    7. To protect the free and open nature of the FreeCAD project.
+
+2. Fundamentals
+
+    1. FreeCAD uses the git distributed revision control system.
+    2. Source code for the main application and related subprojects is hosted on github.com in the FreeCAD organization, hereafter referred to as 'the Platform'.
+    3. Problems are discrete, well-defined limitations or bugs.
+    4. FreeCAD uses the Platform's issue-tracking system to track problems.
```

Comment:
Currently, there are 746 reported issues, of which 387 are closed (359 open), many several years old, and obviously not followed, some actually completed. In the current situation, this doesn't look like a working solution. 

If we want to use the Issues to track problems, there needs to be some active maintaining of them

Reply:
Exactly.  That is precisely the problem we're trying to address with an improved process.  This policy is step one.  Step two is getting the organization to start working that way.  It will take time.  It'll probably hurt like hell.  

Are you proposing a text change to the document or just expressing frustration or concern?

Reply:
My concern is, that one of the main task of a maintainer is to say "No". Many of the existing "Issues" don't have any merit, shouldn't have been opened in the first place. While the C4 is quite precise when talking about patches and PRs, it doesn't say anything about "tracking problems". 

So what shall be done about "Issues" that are meaningless ? Look at Issue #6004 for example.

Reply:
> it doesn't say anything about "tracking problems".

Perhaps this is intentional.  The document does talk about the process being to first open an issue.  It talks about "accurately identifying problems and applying minimal, accurate solutions"
The concept here is that it forces the discussion (and argument) upstream to the issue so that the solutions are less controversial.
Issues that are getting no discussion and also no development attention are candidates to close.  The actual closing is a judgment call as discussed elsewhere.

Reply:
There is no mention of the forum. How about adding a precision here:

> For help questions, feature requests and general discussions, FreeCAD uses the project's web forum

Reply:
C4 and this derived version do not use the term "feature request" at all.  C4 goes even further by prohibiting users from making "feature requests".  To describe new functionality that is wanted, you describe it in terms of a problem - something you can't do with the current software.  

I added a statement about help requests and general discussions.